File: RPi Experiments/RPi2-1/C/08_flowingLeds.c
#include <wiringPi.h>
#include <stdio.h>

//turn on led_n
void led_on(int n)
{
	digitalWrite(n, LOW);
}

//turn off led_n
void led_off(int n)
{
	digitalWrite(n, HIGH);
}

int main(void)
{
    int i;
    int pins[8] = {7, 0, 2, 3, 1, 4, 5, 6}; //wiringPi pin numbers for the LEDs


    if(wiringPiSetup() == -1)
    { 
		printf("setup wiringPi failed !");
		return 1; 
	}

    for(i=0;i<8;i++) //iterate trough all pins
    {       
        pinMode(pins[i], OUTPUT); //initialize the LED pin as output
        led_off(pins[i]);
	}

    while(1)   
    {
        int ms = 200;
        //iterate over the pins
        //turn the led on from lowest to the highest
        for(i=0;i<8;i++)
        {   
			led_on(pins[i]);
			delay(ms); //wait for ms
        }
        //fade from the highest to the lowest
        for(i=8;i>=0;i--)
        {  
            led_off(pins[i]);
			delay(ms); //wait for ms			
        }
        //turn the led on from highest to the lowest
        for(i=8;i>=0;i--)
        {  
            led_on(pins[i]);
			delay(ms); //wait for ms		
        }
        //fade from the lowest to the highest
        for(i=0;i<8;i++)
        {   
			led_off(pins[i]);
			delay(ms); //wait for ms
        }
	}

	return 0;
}

File: RPi Experiments/RPi2-1/Python/08_flowingLeds.py
import RPi.GPIO as GPIO
import time

pins = [7, 11, 13, 15, 12, 16, 18, 22] # physical pin numbers for the LEDs
invertedPins = [22, 18, 16, 12, 15, 13, 11, 7] # physical pin numbers for the LEDs in reverse order

def setup():
    GPIO.setwarnings(False) # don't show warnings if GPIO is in use
    GPIO.setmode(GPIO.BOARD)        # use GPIO by physical location
    for pin in pins:                # iterate throug all pin numbers
        GPIO.setup(pin, GPIO.OUT)   # initialize the LED pin as output
        GPIO.output(pin, GPIO.HIGH) # turn off LED

def loop():
    while True:
        ms = 0.2
        for pin in pins: # turn on LEDs from left to right
            GPIO.output(pin, GPIO.LOW)	
            time.sleep(ms)
        for pin in invertedPins: # turn off LEDs from right to left
            GPIO.output(pin, GPIO.HIGH)
            time.sleep(ms)
        for pin in invertedPins: # turn on LEDs from right to left
            GPIO.output(pin, GPIO.LOW)	
            time.sleep(ms)
        for pin in pins: # turn off LEDs from left to right
            GPIO.output(pin, GPIO.HIGH)	
            time.sleep(ms)

def destroy():
	for pin in pins:
		GPIO.output(pin, GPIO.HIGH)    # turn off all LEDs
	GPIO.cleanup()                     # release resource

if __name__ == '__main__':     # program start from here
	setup()
	try:
		loop()
	except KeyboardInterrupt:  # when 'Ctrl+C' is pressed, the child program destroy() will be  executed.
		destroy()
